packages/xo-web/src/xo-app/sr/tab-general.js

```diff
@@ -32,9 +165,98 @@ export default class TabGeneral extends Component {
     }
   }
 
-  render() {
-    const { sr, vdis, vdiSnapshots, unmanagedVdis } = this.props
+  _getDiskGroups = createSelector(
+    () => this.props.vdis,
+    () => this.props.vdiSnapshots,
+    () => this.props.unmanagedVdis,
+    (vdis, vdiSnapshots, unmanagedVdis) => {
+      const diskParents = []
+      const groups = []
+      const snapshotsByVdi = groupBy(vdiSnapshots, '$snapshot_of')
+      const unmanagedVdisById = keyBy(unmanagedVdis, 'id')
+      let baseCopiesUsage
+      let bc
+      let nBaseCopies
+      let root
+
+      let orphanedVdiSnapshots
+      if ((orphanedVdiSnapshots = snapshotsByVdi[undefined]) !== undefined) {
+        orphanedVdiSnapshots.forEach(snapshot => {
+          groups.push(snapshot)
+        })
+      }
 
+      // search root base copy for each VDI
+      const disks = vdis.map(({ id, parent, snapshots, ...vdi }) => {
+        baseCopiesUsage = 0
+        nBaseCopies = 0
+        root = undefined
+        while ((bc = unmanagedVdisById[parent]) !== undefined) {
+          root = bc.id
+          parent = bc.parent
+          if (!diskParents.includes(root)) {
+            nBaseCopies++
+            baseCopiesUsage += bc.usage
+            diskParents.push(root)
+          }
+        }
+
+        if ((snapshots = snapshotsByVdi[id]) !== undefined) {
+          // snapshot can have base copy without active VDI
+          snapshots.forEach(({ parent }) => {
+            if (parent !== undefined && !diskParents.includes(parent)) {
+              nBaseCopies++
```

Comment:
Idem: store base copies and check if you already have the with the global VDI `Set`.